hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/processor/TransformStatefulP.java

```diff
@@ -154,6 +167,9 @@ void reset(Watermark wm) {
 
     @Override
     public boolean saveToSnapshot() {
+        if (inComplete) {
```

Comment:
why do we need this? we don't this for other stateful processors (i.e. windowing)